blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServerShutdownIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.HashSet;
+
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.*;
+
+public class HttpMetricsIngestionServerShutdownIntegrationTest {
+
+    private static HttpMetricsIngestionServer server;
+    private static HttpClientVendor vendor;
+    private static DefaultHttpClient client;
+    private static Collection<Integer> manageShards = new HashSet<Integer>();
+    private static int httpPort;
+    private static ScheduleContext context;
+    private static EventsIO eventsSearchIO;
+    private static EsSetup esSetup;
+    //A time stamp 2 days ago
+    private final long baseMillis = Calendar.getInstance().getTimeInMillis() - 172800000;
+
+
+    @BeforeClass
+    public static void setUp() throws Exception{
+        System.setProperty(CoreConfig.EVENTS_MODULES.name(), "com.rackspacecloud.blueflood.io.EventElasticSearchIO");
+        Configuration.getInstance().init();
+        httpPort = Configuration.getInstance().getIntegerProperty(HttpConfig.HTTP_INGESTION_PORT);
+        manageShards.add(1); manageShards.add(5); manageShards.add(6);
+        context = spy(new ScheduleContext(System.currentTimeMillis(), manageShards));
+
+        esSetup = new EsSetup();
+        esSetup.execute(EsSetup.deleteAll());
+        esSetup.execute(EsSetup.createIndex(EventElasticSearchIO.EVENT_INDEX)
+                .withSettings(EsSetup.fromClassPath("index_settings.json"))
+                .withMapping("graphite_event", EsSetup.fromClassPath("events_mapping.json")));
+        eventsSearchIO = new EventElasticSearchIO(esSetup.client());
+        server = new HttpMetricsIngestionServer(context, new AstyanaxMetricsWriter());
+        server.setHttpEventsIngestionHandler(new HttpEventsIngestionHandler(eventsSearchIO));
+
+        server.startServer();
+
+        vendor = new HttpClientVendor();
+        client = vendor.getClient();
+    }
+
+    @Test
+    public void testHttpIngestionHappyCase() throws Exception {
+
+        // given
+        HttpPost post = new HttpPost(getMetricsURI());
+        HttpEntity entity = new StringEntity(JSONMetricsContainerTest.generateJSONMetricsData(),
+                ContentType.APPLICATION_JSON);
+        post.setEntity(entity);
+        HttpResponse response = client.execute(post);
+        Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+        HttpPost post2 = new HttpPost(getMetricsURI());
+        HttpEntity entity2 = new StringEntity(JSONMetricsContainerTest.generateJSONMetricsData(),
+                ContentType.APPLICATION_JSON);
+        post2.setEntity(entity2);
+
+        // when
+        server.shutdownServer();
+
+        // then
+        try {
+            HttpResponse response2 = client.execute(post2);
+            Assert.fail("We should have received a Connect exception");
```

Comment:
I originally did something like that, using jUnit's [`ExpectedException`](http://junit.org/javadoc/latest/org/junit/rules/ExpectedException.html) rule. However, `ConnectException` can indicate a number of failure conditions, including timeout on connection.

For example, if the service did not shut down correctly, and the socket is still open, but it's timing out while opening a connection (for whatever reason). That would give us a false pass, because a timeout on a socket connection would yield a `ConnectException`. If our test code is only looking for the exception, and not considering the underlying cause, it would be incorrect. Therefore, we have to look at the exception message to make sure that it actually represents a refused connection; and in order to do that, we need the exception object to assert against, which prevents the use of the `ExpectedException` rule.

Would it help if I added a comment explaining the `try/catch` to future readers?

FWIW, Groovy actually has something that does this, built into the language. You could type:

```
when:
// do something

then:
def ex = notThrown(ConnectException)
ex.message == "Connection refused"
```

It looks much nicer than explicitly catching the exception, can allows you to make assertions on it.